File: Xamarin_Forms/Xamarin_Forms/GBotXamarinForms/GBotXamarinForms/Control/LedUtil.cs
using Android.Runtime;
using System;

namespace GBotXamarinForms
{
    class LedUtil
    {
        public static void SetBlueLed()
        {
            SetLed("0x0200");
        }

        public static void SetBluePlusWhiteLed()
        {
            SetLed("0x0220");
        }

        public static void SetLightBlueLed()
        {
            SetLed("0x0270");
        }

        public static void SetPinkLed()
        {
            SetLed("0x0280");
        }

        public static void SetLightPinkLed()
        {
            SetLed("0x0380");
        }

        public static void SetCyanLed()
        {
            SetLed("0x0300");
        }

        public static void SetGreenLed()
        {
            SetLed("0x0100");
        }

        public static void SetLightGreen()
        {
            SetLed("0x0119");
        }

        public static void SetLightGreenPlusWhite()
        {
            SetLed("0x0140");
        }

        public static void SetRedLed()
        {
            SetLed("0x0080");
        }

        public static void SetOrangeLed()
        {
            SetLed("0x0180");
        }

        public static void SetOnLed()
        {
            SetLed("0xffff");
        }

        public static void SetOffLed()
        {
            SetLed("0x0000");
        }
        public static void SetLed(string cmd)
        {
            try
            {
                // JNI localiza a classe WriteSysFileManager
                IntPtr class_ref = JNIEnv.FindClass("android/os/WriteSysFileManager");

                // JNI para chamar o construtor
                IntPtr id_ctor_I = JNIEnv.GetMethodID(class_ref, "<init>", "()V");

                // JNI Criar uma instancia da classe
                IntPtr lrefInstance = JNIEnv.NewObject(class_ref, id_ctor_I);

                // JNI para localizar o metodo writeSysFile
                IntPtr id_write = JNIEnv.GetStaticMethodID(class_ref, "writeSysFile", "(Ljava/lang/String;Ljava/lang/String;)Z");

                // JNI para chamar o metodo writeSysFile e alterar os parametros do Led via sys
                _ = JNIEnv.CallStaticBooleanMethod(class_ref, id_write, new JValue[2] { new JValue(new Java.Lang.String("/sys/class/leds/aw9109_led/onoff_by_reg")), new JValue(new Java.Lang.String(cmd)) });
            
            } catch(Exception e)
            {
                Console.WriteLine(e.Message);
            }
            
        }
    }
}

File: Android/app/src/main/java/com/gertec/exemplosgertec/service/LedUtil.java
package com.gertec.exemplosgertec.CodigoBarras.service;

import android.nfc.Tag;
import android.util.Log;

import java.lang.reflect.Method;

public class LedUtil {

    private static final String TAG = LedUtil.class.getSimpleName();

    public static void setBlueLed(){
        Log.d(TAG,"luz azul");
        setLed("0x0200");
    }

    public static void setBluePlusWhiteLed(){
        Log.d(TAG,"luz azul + Branco");
        setLed("0x0220");
    }

    public static void setLightBlueLed(){
        Log.d(TAG,"luz azul Claro");
        setLed("0x0270");
    }

    public static void setPinkLed(){
        Log.d(TAG,"luz Rosa");
        setLed("0x0280");
    }

    public static void setLightPinkLed(){
        Log.d(TAG,"luz Rosa Claro");
        setLed("0x0380");
    }

    public static void setCyanLed(){
        Log.d(TAG,"luz Ciano");
        setLed("0x0300");
    }

    public static void setGreenLed(){
        Log.d(TAG,"luz verde");
        setLed("0x0100");
    }

    public static void setLightGreen(){
        Log.d(TAG,"luz Verde Claro");
        setLed("0x0119");
    }

    public static void setLightGreenPlusWhite(){
        Log.d(TAG,"luz Verde Claro + branco");
        setLed("0x0140");
    }

    public static void setRedLed(){
        Log.d(TAG,"luz vermelho");
        setLed("0x0080");
    }

    public static void setOrangeLed(){
        Log.d(TAG,"luz laranja");
        setLed("0x0180");
    }

    public static void setOnLed(){
        Log.d(TAG,"Ligue a luz de digitalização");
        setLed("0xffff");
    }

    public static void setOffLed(){
        Log.d(TAG,"Desligue a luz de digitalização");
        setLed("0x0000");
    }


    private static void setLed(String cmd){
        try{
            Class WriteSysFileManagerClass = Class.forName("android.os.WriteSysFileManager");
            Object WriteSysFileManager = WriteSysFileManagerClass.newInstance();
            Method method = WriteSysFileManagerClass.getMethod("writeSysFile", String.class, String.class);
            method.invoke(WriteSysFileManager, "/sys/class/leds/aw9109_led/onoff_by_reg",cmd);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
